index.js

```diff
@@ -38,22 +26,25 @@ if (hasToStringTag && gOPD && getPrototypeOf) {
 }
 
 var tryTypedArrays = function tryAllTypedArrays(value) {
-	var anyTrue = false;
-	forEach(toStrTags, function (getter, typedArray) {
-		if (!anyTrue) {
-			try {
-				anyTrue = getter.call(value) === typedArray;
-			} catch (e) { /**/ }
-		}
-	});
-	return anyTrue;
+	// note: can update syntax to friendlier for..of when dropping support for node 0.10
+	var keys = Object.keys(toStrTags);
+	for (var i = 0; i < keys.length; i++) {
+		var typedArray = keys[i];
+		var getter = toStrTags[typedArray];
+		try {
+			if (getter.call(value) === typedArray) {
+				return true;
+			}
+		} catch (e) { /**/ }
+	}
+	return false;
 };
 
 module.exports = function isTypedArray(value) {
 	if (!value || typeof value !== 'object') { return false; }
 	if (!hasToStringTag || !(Symbol.toStringTag in value)) {
-		var tag = $slice($toString(value), 8, -1);
-		return $indexOf(typedArrays, tag) > -1;
+		var tag = String.prototype.slice.call(Object.prototype.toString.call(value), 8, -1);
+		return Array.prototype.indexOf.call(typedArrays, tag) > -1;
```

Comment:
Now we’d have an unsafe runtime dependency on Function.prototype.call - which is unacceptable in any JS engine, in any version.